In [1]:
import pandas as pd
import matplotlib.pylab as plt
import re

# 0. Reading the survey data with pandas

In [2]:
data = pd.read_csv('survey_data.csv')

In [3]:
data.head()

타임스탬프 연령대를 선택해주세요. (2020 - "출생년도" + 1의 결과값)  \
0  2020/05/18 5:26:50 오후 GMT+9                              20 ~ 24세   
1  2020/05/18 5:43:14 오후 GMT+9                              20 ~ 24세   
2  2020/05/18 5:44:06 오후 GMT+9                              20 ~ 24세   
3  2020/05/18 5:45:14 오후 GMT+9                                19세 이하   
4  2020/05/18 5:46:18 오후 GMT+9                              20 ~ 24세   

  성별을 선택해주세요 기초제품을 구매하거나 선물한 적이 있나요?  \
0         여성                       예   
1         여성                       예   
2         여성                       예   
3         여성                       예   
4         여성                       예   

                평소 화장품에 대한 관심/지식은 어느정도인가요? (복수응답 가능)  \
0  건성, 복합성, 지성이 무엇인지는 알고 있어요.;비타민이 미백에 좋다는 것, 시카와...   
1  비타민이 미백에 좋다는 것, 시카와 티트리 성분이 진정효과가 있다는 것 등을 알고 ...   
2  건성, 복합성, 지성이 무엇인지는 알고 있어요.;비타민이 미백에 좋다는 것, 시카와...   
3                     자신의 피부에 맞게 상품을 비교하고 선택할 수 있어요.   
4  비타민이 미백에 좋다는 것, 시카와 티트리 성분이 진정효과가 있다는 것 등을 알고 ...   

  기초제품 구매 시, 선택에 어려움을 겪은 적이 있나요? 그 이유를 말씀해주세요. (복수응답가능)  \
0  제품이 너무 다양하고 많아서 정보를 알 수 없어서;진짜 좋은 제품인지 알 수 없어서...      
1     제품이 너무 다양하고 많아서 정보를 알 수 없어서;진짜 좋은 제품인지 알 수 없어서      
2     제품이 너무 다양하고 많아서 정보를 알 수 없어서;진짜 좋은 제품인지 알 수 없어서      
3                                 진짜 좋은 제품인지 알 수 없어서      
4  제품이 너무 다양하고 많아서 정보를 알 수 없어서;피부가 민감하지 않아서 제품 다양...      

           화장품 구매에 있어, 오프라인 매장 방문의 목적이 무엇인가요? (복수응답)  \
0  제품을 직접 테스트하고 크기 등을 보려고;사람에게 물어보고 구매하고 싶어서;온라인 ...   
1                             제품을 직접 테스트하고 크기 등을 보려고   
2          제품을 직접 테스트하고 크기 등을 보려고;사람에게 물어보고 구매하고 싶어서   
3                             제품을 직접 테스트하고 크기 등을 보려고   
4         집이랑 가까워서;그냥 집가다가 생각나면 사는 거고 아니면 쿠팡으로 사는거구~   

                          오프라인 매장 방문을 꺼리는 이유는 무엇인가요?  \
0                               오프라인 매장 방문을 꺼리지 않아요!   
1  오프라인 매장 방문을 꺼리지 않아요!;온라인 쇼핑에 익숙해져서;혼자 비교하고 결정하...   
2                       직원들의 접객이 불편해서;혜택이 더 다양하고 많아서   
3                               오프라인 매장 방문을 꺼리지 않아요!   
4                               오프라인 매장 방문을 꺼리지 않아요!   

   화장품과 관련하여 유튜브로 정보를 탐색하시나요?  화장품과 관련한 어플을 사용하시나요? (화해, 글로우픽 등)  \
0                         1.0                                0.0   
1                         0.0                                0.0   
2                         3.0                                1.0   
3                         4.0                                4.0   
4                         1.0                                0.0   

                화장품과 관련하여 유튜브 혹은 어플을 사용하는 이유는 무엇인가요?  \
0         언젠가 구매할지도 모르니까;색조, 메이트업베이스 등에 대한 정보를 얻기 위해   
1                                            사용하지 않음   
2             유행을 알고 싶어서;색조, 메이트업베이스 등에 대한 정보를 얻기 위해   
3  색조, 메이트업베이스 등에 대한 정보를 얻기 위해;기초제품(스킨, 로션, 미스트, ...   
4                             그냥 화장한 얼굴이 예뻐서 대리만족하려구   

              어떻게 하면 기초 제품 구매가 더 편리할 것 같아요? (복수응답가능)  
0  직원의 대면 응대를 활성화해주세요(상담은 사람과 만나서 하고 싶어요, 추천받은 제품...  
1  비대면 응대를 활성화해주세요(낯을 가려요, 집에서 쇼핑하고 싶어요, 혼자 고민하고 ...  
2  직원의 대면 응대를 활성화해주세요(상담은 사람과 만나서 하고 싶어요, 추천받은 제품...  
3  특징별로 상품에 대한 요약 정리를 해주세요(보습라인 : 가,나,다 / 여드름 진정 ...  
4  특징별로 상품에 대한 요약 정리를 해주세요(보습라인 : 가,나,다 / 여드름 진정 ...

In [4]:
data.shape

(135, 12)

11 questions(excluding time stamp) and 135 answers

# 1. Classifying the questions and the answers as a table

## 1 - 1. Making a question list

Before making the list, I need to translate the questions as English.

The questions can be traslated as follows:

 - Ages
 - Gender
 - Have you ever purchased or present the skin care products? : 'The experience status'
 - How much do you know about the skin care? : 'The knowledge about skin care'
 - Have you ever been in trouble in purchasing the skin care porducts? : 'A difficult experience in purchasing'
 - Why do you visit the offline stores instead online mall? : 'The reasons for visit'
 - Why do you mind visiting the offline stores? Why you prefer to online? : 'The reasons not to visit'
 - Do you usually searching for the information through youtube contents? : 'Youtube usage status'
 - Do you usually searching for the information through related applications? : 'Application usage status'
 - Why do you use the applications or youtube contents for getting the informations about skin care? : 'The reasons for use the apps or youtube'
 - Can you tell me how to improve the offline store systems to make the consumers easy to purchase the skin care products? : 'The way to improve'

In [10]:
# questions without time stamp

ques = list(data.columns)[1:]
ques

['연령대를 선택해주세요. (2020 - "출생년도" + 1의 결과값)',
 '성별을 선택해주세요',
 '기초제품을 구매하거나 선물한 적이 있나요?',
 '평소 화장품에 대한 관심/지식은 어느정도인가요? (복수응답 가능)',
 '기초제품 구매 시, 선택에 어려움을 겪은 적이 있나요? 그 이유를 말씀해주세요. (복수응답가능)',
 '화장품 구매에 있어, 오프라인 매장 방문의 목적이 무엇인가요? (복수응답)',
 '오프라인 매장 방문을 꺼리는 이유는 무엇인가요?',
 '화장품과 관련하여 유튜브로 정보를 탐색하시나요?',
 '화장품과 관련한 어플을 사용하시나요? (화해, 글로우픽 등)',
 '화장품과 관련하여 유튜브 혹은 어플을 사용하는 이유는 무엇인가요?',
 '어떻게 하면 기초 제품 구매가 더 편리할 것 같아요? (복수응답가능)']

## 1 - 2. Making an answer sheet : gathering the answers including each opinions

Let's check the answers with multiple choices.

Befor that, make

In [6]:
data.iloc[:, 5]

0      제품이 너무 다양하고 많아서 정보를 알 수 없어서;진짜 좋은 제품인지 알 수 없어서...
1         제품이 너무 다양하고 많아서 정보를 알 수 없어서;진짜 좋은 제품인지 알 수 없어서
2         제품이 너무 다양하고 많아서 정보를 알 수 없어서;진짜 좋은 제품인지 알 수 없어서
3                                     진짜 좋은 제품인지 알 수 없어서
4      제품이 너무 다양하고 많아서 정보를 알 수 없어서;피부가 민감하지 않아서 제품 다양...
5                                     진짜 좋은 제품인지 알 수 없어서
6                            제품이 너무 다양하고 많아서 정보를 알 수 없어서
7      제품이 너무 다양하고 많아서 정보를 알 수 없어서;매번 실패해서;진짜 좋은 제품인지...
8         제품이 너무 다양하고 많아서 정보를 알 수 없어서;진짜 좋은 제품인지 알 수 없어서
9       진짜 좋은 제품인지 알 수 없어서;검색이 귀찮아서;더 싸게 살 수 있을 것같아서 미뤄서
10        제품이 너무 다양하고 많아서 정보를 알 수 없어서;진짜 좋은 제품인지 알 수 없어서
11     피부(스킨케어)에 대한 지식이 없어서;제품이 너무 다양하고 많아서 정보를 알 수 없...
12                                    진짜 좋은 제품인지 알 수 없어서
13                           제품이 너무 다양하고 많아서 정보를 알 수 없어서
14     피부(스킨케어)에 대한 지식이 없어서;제품이 너무 다양하고 많아서 정보를 알 수 없...
15        제품이 너무 다양하고 많아서 정보를 알 수 없어서;진짜 좋은 제품인지 알 수 없어서
16      피부(스킨케어)에 대한 지식이 없어서;제품이 너무 다양하고 많아서 정보를 알 수 없어서
17     제품이 너무 다양하고 많아서 정보를 알 수 

## (보류) 1 - 3. 기타 의견 수집하기

더 쉽게 하는 방법 생각해보기

In [16]:
ans_age = list(data.iloc[:, 1].unique())
ans_gender = list(data.iloc[:, 2].unique())
ans_purchase_experience_status = list(data.iloc[:, 3].unique())
ans_skincare_knowledge = list(data.iloc[:, 4].unique())
ans_trouble_in_purchase = list(data.iloc[:, 5].unique())
ans_reason_visiting_store = list(data.iloc[:, 6].unique())
ans_reason_mind_visiting = list(data.iloc[:, 7].unique())
ans_info_from_app = list(data.iloc[:, 8].unique())
ans_info_from_youtube = list(data.iloc[:, 9].unique())
ans_reason_app_youtube = list(data.iloc[:, 10].unique())
ans_how_to_improve = list(data.iloc[:, 11].unique())

In [17]:
# Gathering all the answers

answers = [ans_age, ans_gender, ans_purchase_experience_status, ans_skincare_knowledge, ans_trouble_in_purchase, ans_reason_visiting_store,
           ans_reason_mind_visiting, ans_info_from_app, ans_info_from_youtube, ans_reason_app_youtube, ans_how_to_improve]

From question number 4, all the questions includes multiple choices.

In [10]:
for ans in ans_skincare_knowledge:
    if ''

SyntaxError: invalid syntax (<ipython-input-10-b867ddace643>, line 2)

In [ ]:
len(ans_skincare_knowledge)

## 1 - 4. (보류) Deleting the doubled option answers

The questions able to submit the opinions are as follows:
 - Have you ever been in trouble in purchasing the skin care porducts? : 'A difficult experience in purchasing'
 - Why do you visit the offline stores instead online mall? : 'The reasons for visit'
 - Why do you mind visiting the offline stores? Why you prefer to online? : 'The reasons not to visit'
 - Why do you use the applications or youtube contents for getting the informations about skin care? : 'The reasons for use the apps or youtube'
 - Can you tell me how to improve the offline store systems to make the consumers easy to purchase the skin care products? : 'The way to improve'
 
So I need to check the answers whether there are the doubled or not.

In [18]:
ans_trouble_in_purchase

['제품이 너무 다양하고 많아서 정보를 알 수 없어서;진짜 좋은 제품인지 알 수 없어서;검색이 귀찮아서',
 '제품이 너무 다양하고 많아서 정보를 알 수 없어서;진짜 좋은 제품인지 알 수 없어서',
 '진짜 좋은 제품인지 알 수 없어서',
 '제품이 너무 다양하고 많아서 정보를 알 수 없어서;피부가 민감하지 않아서 제품 다양성이 주는 혼란이 있지만, 사실 그렇게 신경쓰지 않고 사는 편(그냥 적당한 가격대에 로드샵 대표 스킨케어라인 산다)',
 '제품이 너무 다양하고 많아서 정보를 알 수 없어서',
 '제품이 너무 다양하고 많아서 정보를 알 수 없어서;매번 실패해서;진짜 좋은 제품인지 알 수 없어서',
 '진짜 좋은 제품인지 알 수 없어서;검색이 귀찮아서;더 싸게 살 수 있을 것같아서 미뤄서',
 '피부(스킨케어)에 대한 지식이 없어서;제품이 너무 다양하고 많아서 정보를 알 수 없어서;검색이 귀찮아서',
 '피부(스킨케어)에 대한 지식이 없어서;제품이 너무 다양하고 많아서 정보를 알 수 없어서;진짜 좋은 제품인지 알 수 없어서',
 '피부(스킨케어)에 대한 지식이 없어서;제품이 너무 다양하고 많아서 정보를 알 수 없어서',
 '제품이 너무 다양하고 많아서 정보를 알 수 없어서;(오프라인의 경우)직원이 불친절해서 / 혹은 접객이 미흡해서;진짜 좋은 제품인지 알 수 없어서;검색이 귀찮아서',
 '제품이 너무 다양하고 많아서 정보를 알 수 없어서;진짜 좋은 제품인지 알 수 없어서;검색이 귀찮아서;더 싸게 살 수 있을 것같아서 미뤄서',
 nan,
 '검색이 귀찮아서',
 '피부(스킨케어)에 대한 지식이 없어서',
 '피부(스킨케어)에 대한 지식이 없어서;제품이 너무 다양하고 많아서 정보를 알 수 없어서;매번 실패해서;(오프라인의 경우)직원이 불친절해서 / 혹은 접객이 미흡해서;진짜 좋은 제품인지 알 수 없어서;검색이 귀찮아서;더 싸게 살 수 있을 것같아서 미뤄서',
 '피부(스킨케어)에 대한 지식이 없어서;진짜 좋은 제품인지 알 수 없어서',
 '피부(스킨케어)에 대

# 2. Classifying the respondents : as customers, purchaser and the related.

1. related : The respondents who answered '업계에 종사하고 있어요.'
2. customer : The respondents who did not answer '업계에 종사하고 있어요.'
3. purchasers : The respondents who have ever purchased any skin care products

In [3]:
related = pd.DataFrame(columns = data.columns)
customers =  pd.DataFrame(columns = data.columns)
purchasers =  pd.DataFrame(columns = data.columns)

In [4]:
for i, ans in enumerate(data.iloc[:, 4]):
    if type(ans) is not str:    # the respendents who have never purchased the related products - ignore
        customers = customers.append(data.iloc[i])
        continue
    purchasers = purchasers.append(data.iloc[i])
    if '업계에 종사하고 있어요.' in ans:
        related = related.append(data.iloc[i])
    else:
        customers = customers.append(data.iloc[i])

In [5]:
related.shape

(3, 12)

In [6]:
customers.shape

(132, 12)

In [7]:
purchasers.shape

(122, 12)

# 3. Recognition of offline/online store

## 3 - 1. The reasons why the customers mind visiting offline store

오프라인 매장 방문을 꺼리는 이유 살피기

In [11]:
answers_count = dict()
answers = list(purchasers.iloc[:, 7])

for answer in answers:
    if ';' in answer:
        for ans in answer.split(';'):
            if ans not in answers_count.keys():
                answers_count[ans] = 1
            else:
                answers_count[ans] += 1
    else:
        if answer not in answers_count.keys():
            answers_count[answer] = 1
        else:
            answers_count[answer] += 1
answers_count

{'오프라인 매장 방문을 꺼리지 않아요!': 58,
 '온라인 쇼핑에 익숙해져서': 20,
 '혼자 비교하고 결정하고 싶어서': 20,
 '직원들의 접객이 불편해서': 28,
 '혜택이 더 다양하고 많아서': 16,
 '집에서 나가고 싶지 않아서': 15,
 '온라인이 더 싸요': 1,
 '어플, 유튜브, 방송 등 스스로도 충분히 정보를 얻을 수 있어서': 18,
 '사람 많은 곳을 좋아하지 않아서': 1,
 '귀찮아서': 1,
 '온라인으로 더 저렴히 구매가능해서': 1,
 '온라인으로 받을 수 있는 혜택(ex.할인)이 많아서': 1,
 '나가기귀찮아서': 1}

오프라인 방문을 꺼리지 않는다고 답한 사람들이 꺼리는 이유를 선택한 경우도 있었다.

선택의 이유는 아래의 경우라고 추정할 수 있다.

1. 본인은 오프라인 방문을 꺼리지 않지만, 사람들이 생각한 이유에 대해서 추측한 답변
2. 오프라인 매장 방문을 꺼리지는 않지만, 한 번씩 불편함을 느끼는 정도
3. 업계 종사자가 느낀 일반 고객들의 생각
4. 기타

중복된 답변은 아래와 같이 처리하자.

1. '귀찮'이라는 답변이 들어간 내용은 '집에서 나가고 싶지 않아서'로 합친다.
2. '온라인', '혜택', '저렴'이라는 단어가 들어간 답변은 '온라인에서의 혜택이 더 다양하고 많아서'라는 답변으로 고친다.
3. 기타(추가 사항이 있다면 기록해두기)

In [12]:
import re

for key, value in answers_count.items():
    if '귀찮' in key:
#         print(key)
        answers_count['집에서 나가고 싶지 않아서'] += value
        answers_count[key] = 0
    elif re.search('온라인.+(혜택|싸[요다서]|저렴)', key):
#         print(key)
        answers_count['혜택이 더 다양하고 많아서'] += value
        answers_count[key] = 0

answers_count

{'오프라인 매장 방문을 꺼리지 않아요!': 58,
 '온라인 쇼핑에 익숙해져서': 20,
 '혼자 비교하고 결정하고 싶어서': 20,
 '직원들의 접객이 불편해서': 28,
 '혜택이 더 다양하고 많아서': 19,
 '집에서 나가고 싶지 않아서': 17,
 '온라인이 더 싸요': 0,
 '어플, 유튜브, 방송 등 스스로도 충분히 정보를 얻을 수 있어서': 18,
 '사람 많은 곳을 좋아하지 않아서': 1,
 '귀찮아서': 0,
 '온라인으로 더 저렴히 구매가능해서': 0,
 '온라인으로 받을 수 있는 혜택(ex.할인)이 많아서': 0,
 '나가기귀찮아서': 0}

In [13]:
# dictionary[new_key] = dictionary.pop(old_key)
answers_count['온라인 구매가 혜택이 더 다양하고 많아서'] = answers_count.pop('혜택이 더 다양하고 많아서')
answers_count

{'오프라인 매장 방문을 꺼리지 않아요!': 58,
 '온라인 쇼핑에 익숙해져서': 20,
 '혼자 비교하고 결정하고 싶어서': 20,
 '직원들의 접객이 불편해서': 28,
 '집에서 나가고 싶지 않아서': 17,
 '온라인이 더 싸요': 0,
 '어플, 유튜브, 방송 등 스스로도 충분히 정보를 얻을 수 있어서': 18,
 '사람 많은 곳을 좋아하지 않아서': 1,
 '귀찮아서': 0,
 '온라인으로 더 저렴히 구매가능해서': 0,
 '온라인으로 받을 수 있는 혜택(ex.할인)이 많아서': 0,
 '나가기귀찮아서': 0,
 '온라인 구매가 혜택이 더 다양하고 많아서': 19}

In [14]:
answers = dict()

for k, v in answers_count.items():
    if v:
        answers[k] = v

answers

{'오프라인 매장 방문을 꺼리지 않아요!': 58,
 '온라인 쇼핑에 익숙해져서': 20,
 '혼자 비교하고 결정하고 싶어서': 20,
 '직원들의 접객이 불편해서': 28,
 '집에서 나가고 싶지 않아서': 17,
 '어플, 유튜브, 방송 등 스스로도 충분히 정보를 얻을 수 있어서': 18,
 '사람 많은 곳을 좋아하지 않아서': 1,
 '온라인 구매가 혜택이 더 다양하고 많아서': 19}

In [32]:
'''
오프라인 매장 방문을 조금이라도 꺼리는 사람들의 비율
     = "오프라인 방문을 꺼리지 않아요"를 선택하지 않은 사람들
        + 꺼리지 않지만, 이유는 알겠는 사람들
구매 경험이 있는 사람들(purchaser) 중에서 살피기
'''

num_mind_visiting, num_prefer_online, num_dislike_hospitality = 0, 0, 0

for ans in purchasers.iloc[:, 7]:
    if re.search('오프라인 매장 방문을 꺼리지 않아요!', ans):
        if ';' in ans:
            print(ans.split(';')[1:])
        else:
            continue
    num_mind_visiting += 1
    if re.search('온라인', ans):
        num_prefer_online += 1
    elif re.search('(혼자|직원(접객)?)', ans):
        num_dislike_hospitality += 1
        
print(f'오프라인 매장 방문을 조금이라도 꺼리는 고객의 비율 : {round(num_mind_visiting*100/purchasers.shape[0], 2)}%')
print(f'온라인을 선호하기 떄문에 굳이 매장에 오지 않는 고객의 비율 : {round(num_prefer_online*100/num_mind_visiting, 2)}%') # 오프라인을 조금이라도 꺼리는 사람들 중 온라인을 선호기 때문에 굳이 오프라인에 방문하지 않는 비율 찾기
print(f'접객이 불편하거나 혼자 의사결정을 하고 싶은 고객의 비율 : {round(num_dislike_hospitality*100/num_mind_visiting, 2)}%')

['온라인 쇼핑에 익숙해져서', '혼자 비교하고 결정하고 싶어서']
['혜택이 더 다양하고 많아서']
['직원들의 접객이 불편해서']
['온라인 쇼핑에 익숙해져서', '혜택이 더 다양하고 많아서', '어플, 유튜브, 방송 등 스스로도 충분히 정보를 얻을 수 있어서']
['귀찮아서']
['직원들의 접객이 불편해서']
['온라인으로 받을 수 있는 혜택(ex.할인)이 많아서']
['어플, 유튜브, 방송 등 스스로도 충분히 정보를 얻을 수 있어서']
['집에서 나가고 싶지 않아서']
['온라인 쇼핑에 익숙해져서']
오프라인 매장 방문을 조금이라도 꺼리는 고객의 비율 : 60.66%
온라인을 선호하기 떄문에 굳이 매장에 오지 않는 고객의 비율 : 31.08%
접객이 불편하거나 혼자 의사결정을 하고 싶은 고객의 비율 : 44.59%


접객이 불편하거나 혼자 의사결정을 하고 싶은 고객의 정보 좀 더 살피기

1. 연령대
2. 성별
3. 후보군 추천 희망 비율

In [16]:
print(num_dislike_hospitality)

33


In [17]:
purchasers.head(3)

타임스탬프 연령대를 선택해주세요. (2020 - "출생년도" + 1의 결과값)  \
0  2020/05/18 5:26:50 오후 GMT+9                              20 ~ 24세   
1  2020/05/18 5:43:14 오후 GMT+9                              20 ~ 24세   
2  2020/05/18 5:44:06 오후 GMT+9                              20 ~ 24세   

  성별을 선택해주세요 기초제품을 구매하거나 선물한 적이 있나요?  \
0         여성                       예   
1         여성                       예   
2         여성                       예   

                평소 화장품에 대한 관심/지식은 어느정도인가요? (복수응답 가능)  \
0  건성, 복합성, 지성이 무엇인지는 알고 있어요.;비타민이 미백에 좋다는 것, 시카와...   
1  비타민이 미백에 좋다는 것, 시카와 티트리 성분이 진정효과가 있다는 것 등을 알고 ...   
2  건성, 복합성, 지성이 무엇인지는 알고 있어요.;비타민이 미백에 좋다는 것, 시카와...   

  기초제품 구매 시, 선택에 어려움을 겪은 적이 있나요? 그 이유를 말씀해주세요. (복수응답가능)  \
0  제품이 너무 다양하고 많아서 정보를 알 수 없어서;진짜 좋은 제품인지 알 수 없어서...      
1     제품이 너무 다양하고 많아서 정보를 알 수 없어서;진짜 좋은 제품인지 알 수 없어서      
2     제품이 너무 다양하고 많아서 정보를 알 수 없어서;진짜 좋은 제품인지 알 수 없어서      

           화장품 구매에 있어, 오프라인 매장 방문의 목적이 무엇인가요? (복수응답)  \
0  제품을 직접 테스트하고 크기 등을 보려고;사람에게 물어보고 구매하고 싶어서;온라인 ...   
1                             제품을 직접 테스트하고 크기 등을 보려고   
2          제품을 직접 테스트하고 크기 등을 보려고;사람에게 물어보고 구매하고 싶어서   

                          오프라인 매장 방문을 꺼리는 이유는 무엇인가요?  \
0                               오프라인 매장 방문을 꺼리지 않아요!   
1  오프라인 매장 방문을 꺼리지 않아요!;온라인 쇼핑에 익숙해져서;혼자 비교하고 결정하...   
2                       직원들의 접객이 불편해서;혜택이 더 다양하고 많아서   

   화장품과 관련하여 유튜브로 정보를 탐색하시나요?  화장품과 관련한 어플을 사용하시나요? (화해, 글로우픽 등)  \
0                         1.0                                0.0   
1                         0.0                                0.0   
2                         3.0                                1.0   

         화장품과 관련하여 유튜브 혹은 어플을 사용하는 이유는 무엇인가요?  \
0  언젠가 구매할지도 모르니까;색조, 메이트업베이스 등에 대한 정보를 얻기 위해   
1                                     사용하지 않음   
2      유행을 알고 싶어서;색조, 메이트업베이스 등에 대한 정보를 얻기 위해   

              어떻게 하면 기초 제품 구매가 더 편리할 것 같아요? (복수응답가능)  
0  직원의 대면 응대를 활성화해주세요(상담은 사람과 만나서 하고 싶어요, 추천받은 제품...  
1  비대면 응대를 활성화해주세요(낯을 가려요, 집에서 쇼핑하고 싶어요, 혼자 고민하고 ...  
2  직원의 대면 응대를 활성화해주세요(상담은 사람과 만나서 하고 싶어요, 추천받은 제품...

7번째 질문에서 혼자 의사결정 혹은 접객 불편을 선택했다면

그사람의 연령대, 성별, 어떻게 하면~의 응답(11)을 기록

In [18]:
ques

['연령대를 선택해주세요. (2020 - "출생년도" + 1의 결과값)',
 '성별을 선택해주세요',
 '기초제품을 구매하거나 선물한 적이 있나요?',
 '평소 화장품에 대한 관심/지식은 어느정도인가요? (복수응답 가능)',
 '기초제품 구매 시, 선택에 어려움을 겪은 적이 있나요? 그 이유를 말씀해주세요. (복수응답가능)',
 '화장품 구매에 있어, 오프라인 매장 방문의 목적이 무엇인가요? (복수응답)',
 '오프라인 매장 방문을 꺼리는 이유는 무엇인가요?',
 '화장품과 관련하여 유튜브로 정보를 탐색하시나요?',
 '화장품과 관련한 어플을 사용하시나요? (화해, 글로우픽 등)',
 '화장품과 관련하여 유튜브 혹은 어플을 사용하는 이유는 무엇인가요?',
 '어떻게 하면 기초 제품 구매가 더 편리할 것 같아요? (복수응답가능)']

In [19]:
df_purchasers = purchasers[[ques[0], ques[1], ques[6], ques[10]]]
df_purchasers.head()

연령대를 선택해주세요. (2020 - "출생년도" + 1의 결과값) 성별을 선택해주세요  \
0                              20 ~ 24세         여성   
1                              20 ~ 24세         여성   
2                              20 ~ 24세         여성   
3                                19세 이하         여성   
4                              20 ~ 24세         여성   

                          오프라인 매장 방문을 꺼리는 이유는 무엇인가요?  \
0                               오프라인 매장 방문을 꺼리지 않아요!   
1  오프라인 매장 방문을 꺼리지 않아요!;온라인 쇼핑에 익숙해져서;혼자 비교하고 결정하...   
2                       직원들의 접객이 불편해서;혜택이 더 다양하고 많아서   
3                               오프라인 매장 방문을 꺼리지 않아요!   
4                               오프라인 매장 방문을 꺼리지 않아요!   

              어떻게 하면 기초 제품 구매가 더 편리할 것 같아요? (복수응답가능)  
0  직원의 대면 응대를 활성화해주세요(상담은 사람과 만나서 하고 싶어요, 추천받은 제품...  
1  비대면 응대를 활성화해주세요(낯을 가려요, 집에서 쇼핑하고 싶어요, 혼자 고민하고 ...  
2  직원의 대면 응대를 활성화해주세요(상담은 사람과 만나서 하고 싶어요, 추천받은 제품...  
3  특징별로 상품에 대한 요약 정리를 해주세요(보습라인 : 가,나,다 / 여드름 진정 ...  
4  특징별로 상품에 대한 요약 정리를 해주세요(보습라인 : 가,나,다 / 여드름 진정 ...

In [20]:
df_purchasers.iloc[0, 0]

'20 ~ 24세'

In [24]:
ages = {}
gender = {'여성' : 0, '남성' : 0, '기타' : 0, '응답을 원하지 않음' : 0}
like_recomendation_sys = 0
basic_ways = [
    '직원의 대면 응대를 활성화해주세요(상담은 사람과 만나서 하고 싶어요, 추천받은 제품을 사는 것이 마음 편해요 등)',
'비대면 응대를 활성화해주세요(낯을 가려요, 집에서 쇼핑하고 싶어요, 혼자 고민하고 싶어요 등)',
'특징별로 상품에 대한 요약 정리를 해주세요(보습라인 : 가,나,다 / 여드름 진정 : A,B,C / 미백라인 : x,y,z 등)',
'응답을 원하지 않아요']
other_ways = []

for age in list(df_purchasers.iloc[:, 0].unique()):
    ages[age] = 0
    
i = 0
for reason in df_purchasers.iloc[:, 2]:
    if re.search('(혼자|직원(접객)?)', reason):
        ages[df_purchasers.iloc[i, 0]] += 1
        gender[df_purchasers.iloc[i, 1]] += 1
        
        if re.search('(요약|정리)', df_purchasers.iloc[i, 3]):
            like_recomendation_sys += 1
    i += 1
    
    
basic_ways = [
    '직원의 대면 응대를 활성화해주세요(상담은 사람과 만나서 하고 싶어요, 추천받은 제품을 사는 것이 마음 편해요 등)',
'비대면 응대를 활성화해주세요(낯을 가려요, 집에서 쇼핑하고 싶어요, 혼자 고민하고 싶어요 등)',
'특징별로 상품에 대한 요약 정리를 해주세요(보습라인 : 가,나,다 / 여드름 진정 : A,B,C / 미백라인 : x,y,z 등)',
'응답을 원하지 않아요']
other_ways = []
for way in df_purchasers.iloc[:, 3]:
    if ';' in way:
        for w in way.split(';'):
            if w not in basic_ways:
                other_ways.append(w)
    else:
        if way not in basic_ways:
            other_ways.append(way)

In [25]:
ages

{'20 ~ 24세': 21,
 '19세 이하': 6,
 '25 ~ 29세': 9,
 '40 ~ 44세': 2,
 '30 ~ 34세': 2,
 '35 ~ 39세': 0,
 '응답을 원하지 않음': 0,
 '45 ~ 49세': 0,
 '50 ~ 59세': 1}

In [26]:
gender

{'여성': 34, '남성': 7, '기타': 0, '응답을 원하지 않음': 0}

In [27]:
like_recomendation_sys

35

In [28]:
other_ways

['사실 특징별로 상품요약은 잘되어있는편이라고 생각하는데 특히 화장품업계에서 무분별한 광고가 오히려 소비자들에게 혼란을 준다고 생각합니당',
 "보통 하나의 브랜드별로는 분류 및 표기가 잘 되어있는데 사실 매장가서 고민하고 검색하는건 여러 브랜드간의 여드름진정라인 비교를 하는것 같아요. 직원분께 추천받는것도 한 브랜드내에서 제품군을 비교받기보단 모든 브랜드 통틀어서 내가 원하는 성분을 포함한 것 중 가성비가 좋거나 '효과'좋은 걸 추천받고 싶으니까요!",
 '특징별 요약 + 상품별 비교가 잘 되어있으면 편할 거 같습니다',
 '알레르기 유발성분을 정리 해주세요']

In [38]:
like_recomendation_sys_whole = 0

for ans in df_purchasers.iloc[:, 3]:
    if re.search('(요약|정리)', ans):
        like_recomendation_sys_whole += 1

print(f'전체 응답자 중에서 요약이나 정리를 선호하는 사람들은 {like_recomendation_sys_whole}명으로, 약 {round(like_recomendation_sys_whole*100/df_purchasers.shape[0], 2)}%이다.')

전체 응답자 중에서 요약이나 정리를 선호하는 사람들은 102명으로, 약 83.61%이다.


In [39]:
35/42

0.8333333333333334

In [40]:
df_purchasers.head(3)

연령대를 선택해주세요. (2020 - "출생년도" + 1의 결과값) 성별을 선택해주세요  \
0                              20 ~ 24세         여성   
1                              20 ~ 24세         여성   
2                              20 ~ 24세         여성   

                          오프라인 매장 방문을 꺼리는 이유는 무엇인가요?  \
0                               오프라인 매장 방문을 꺼리지 않아요!   
1  오프라인 매장 방문을 꺼리지 않아요!;온라인 쇼핑에 익숙해져서;혼자 비교하고 결정하...   
2                       직원들의 접객이 불편해서;혜택이 더 다양하고 많아서   

              어떻게 하면 기초 제품 구매가 더 편리할 것 같아요? (복수응답가능)  
0  직원의 대면 응대를 활성화해주세요(상담은 사람과 만나서 하고 싶어요, 추천받은 제품...  
1  비대면 응대를 활성화해주세요(낯을 가려요, 집에서 쇼핑하고 싶어요, 혼자 고민하고 ...  
2  직원의 대면 응대를 활성화해주세요(상담은 사람과 만나서 하고 싶어요, 추천받은 제품...

In [45]:
df_purchasers_genderless = df_purchasers[[ques[0], ques[6], ques[10]]]

In [55]:
df_purchasers.iloc[0,:][2]

'오프라인 매장 방문을 꺼리지 않아요!'

In [60]:
age_20s = 0; age_40_over = 0
dislike_contact_20s = 0; dislike_contact_40s = 0
prefer_untact_20s = 0; prefer_untact_40s = 0

i = 0
for age in df_purchasers_genderless.iloc[:, 0]:
    if re.search('^2', age):
        age_20s += 1
        if re.search('(접객|혼자)', df_purchasers_genderless.iloc[i, 1]):
            dislike_contact_20s += 1
        if re.search('(비대면|요약|정리)', df_purchasers_genderless.iloc[i, 2]):
            prefer_untact_20s += 1
    elif re.search('^[456]', age):
        age_40_over += 1
        if re.search('(접객|혼자)', df_purchasers_genderless.iloc[i, 1]):
            dislike_contact_40s += 1
        if re.search('(비대면|요약|정리)', df_purchasers_genderless.iloc[i, 2]):
            prefer_untact_40s += 1
    i += 1

print(f'접객이 불편하거나 혼자 보고 싶은 20대는 {dislike_contact_20s}명으로 20대 중 약 {round(dislike_contact_20s*100/age_20s, 2)}%를 차지한다.')
print(f'또한, 비대면 응대를 선호하거나 상품의 요약 정리를 원하는 20대는 {prefer_untact_20s}명으로 20대 중 약 {round(prefer_untact_20s*100/age_20s, 2)}%를 차지한다.')
print(f'접객이 불편하거나 혼자 보고 싶은 40대는 {dislike_contact_40s}명으로 40대 중 약 {round(dislike_contact_40s*100/age_40_over, 2)}%를 차지한다.')
print(f'또한, 비대면 응대를 선호하거나 상품의 요약 정리를 원하는 40대는 {prefer_untact_40s}명으로 40대 중 약 {round(prefer_untact_40s*100/age_40_over, 2)}%를 차지한다.')

접객이 불편하거나 혼자 보고 싶은 20대는 30명으로 20대 중 약 37.5%를 차지한다.
또한, 비대면 응대를 선호하거나 상품의 요약 정리를 원하는 20대는 74명으로 20대 중 약 92.5%를 차지한다.
접객이 불편하거나 혼자 보고 싶은 40대는 3명으로 40대 중 약 18.75%를 차지한다.
또한, 비대면 응대를 선호하거나 상품의 요약 정리를 원하는 40대는 12명으로 40대 중 약 75.0%를 차지한다.


오프라인 매장 방문 목적

In [83]:
reason_visit = {}
reason_visit['급해서 혹은 빨리 사용하고 싶어서'] = 0
reason_visit['온라인은 신뢰할 수 없어서'] = 0

for reason in purchasers.iloc[:,6]:
    if '방문하지 않아요' == reason:
        continue
    elif ';' in reason:
        for r in reason.split(';'):
            if r == '방문하지 않아요':
                continue
            elif re.search('(급해서|당장|빨리)', r):
#                 print(r)
                reason_visit['급해서 혹은 빨리 사용하고 싶어서'] += 1
                reason_visit[r] = 0
                continue
            elif re.search('온라인.*(신뢰|믿을\s?수)', r):
                reason_visit['온라인은 신뢰할 수 없어서'] += 1
                reason_visit[r] = 0
                continue
            if r not in reason_visit.keys():
                reason_visit[r] = 1
            else:
                reason_visit[r] += 1

reason_visit

{'급해서 혹은 빨리 사용하고 싶어서': 3,
 '온라인은 신뢰할 수 없어서': 2,
 '제품을 직접 테스트하고 크기 등을 보려고': 43,
 '사람에게 물어보고 구매하고 싶어서': 30,
 '온라인 쇼핑이 익숙하지 않아서': 8,
 '온라인 결제가 귀찮아서/불편해서': 15,
 '집이랑 가까워서': 24,
 '그냥 집가다가 생각나면 사는 거고 아니면 쿠팡으로 사는거구~': 1,
 '온라인은 신뢰가 잘 되지 않는 느낌이라서요. 인스타 광고 제품이 많은데 믿을 만한 제품인지 잘 모르겠어서': 0,
 '샘플, 증정품 등 혜택이 더 다양하고 많아서': 10,
 '급해서': 0,
 '지금 당장 필요해서': 0,
 '빨리쓰고싶어서 지금당장!': 0,
 '온라인은 믿을 수 없어서': 0}

In [88]:
reason_visit_store = dict()

for k, v in reason_visit.items():
    if re.search('(그냥|생각이?/s나면)', k):
        reason_visit_store['특별한 이유 없음'] = v
        continue
    elif v:
        reason_visit_store[k] = v
        
reason_visit_store

{'급해서 혹은 빨리 사용하고 싶어서': 3,
 '온라인은 신뢰할 수 없어서': 2,
 '제품을 직접 테스트하고 크기 등을 보려고': 43,
 '사람에게 물어보고 구매하고 싶어서': 30,
 '온라인 쇼핑이 익숙하지 않아서': 8,
 '온라인 결제가 귀찮아서/불편해서': 15,
 '집이랑 가까워서': 24,
 '특별한 이유 없음': 1,
 '샘플, 증정품 등 혜택이 더 다양하고 많아서': 10}

In [89]:
related

타임스탬프 연령대를 선택해주세요. (2020 - "출생년도" + 1의 결과값)  \
0   2020/05/18 5:26:50 오후 GMT+9                              20 ~ 24세   
22  2020/05/18 6:21:54 오후 GMT+9                              30 ~ 34세   
62  2020/05/18 7:58:14 오후 GMT+9                              20 ~ 24세   

   성별을 선택해주세요 기초제품을 구매하거나 선물한 적이 있나요?  \
0          여성                       예   
22         여성                       예   
62         남성                       예   

                 평소 화장품에 대한 관심/지식은 어느정도인가요? (복수응답 가능)  \
0   건성, 복합성, 지성이 무엇인지는 알고 있어요.;비타민이 미백에 좋다는 것, 시카와...   
22  건성, 복합성, 지성이 무엇인지는 알고 있어요.;비타민이 미백에 좋다는 것, 시카와...   
62                                      업계에 종사하고 있어요.   

   기초제품 구매 시, 선택에 어려움을 겪은 적이 있나요? 그 이유를 말씀해주세요. (복수응답가능)  \
0   제품이 너무 다양하고 많아서 정보를 알 수 없어서;진짜 좋은 제품인지 알 수 없어서...      
22                                           검색이 귀찮아서      
62  피부(스킨케어)에 대한 지식이 없어서;제품이 너무 다양하고 많아서 정보를 알 수 없...      

            화장품 구매에 있어, 오프라인 매장 방문의 목적이 무엇인가요? (복수응답)  \
0   제품을 직접 테스트하고 크기 등을 보려고;사람에게 물어보고 구매하고 싶어서;온라인 ...   
22                   온라인 배송 시 일회용 쓰레기가 생겨서 - 환경주의자입니다   
62  제품을 직접 테스트하고 크기 등을 보려고;사람에게 물어보고 구매하고 싶어서;집이랑 ...   

   오프라인 매장 방문을 꺼리는 이유는 무엇인가요?  화장품과 관련하여 유튜브로 정보를 탐색하시나요?  \
0        오프라인 매장 방문을 꺼리지 않아요!                         1.0   
22       오프라인 매장 방문을 꺼리지 않아요!                         0.0   
62       오프라인 매장 방문을 꺼리지 않아요!                         2.0   

    화장품과 관련한 어플을 사용하시나요? (화해, 글로우픽 등)  \
0                                 0.0   
22                                0.0   
62                                3.0   

          화장품과 관련하여 유튜브 혹은 어플을 사용하는 이유는 무엇인가요?  \
0   언젠가 구매할지도 모르니까;색조, 메이트업베이스 등에 대한 정보를 얻기 위해   
22                                     사용하지 않음   
62      기초제품(스킨, 로션, 미스트, 선케어 등)에 대한 정보를 얻기 위해   

               어떻게 하면 기초 제품 구매가 더 편리할 것 같아요? (복수응답가능)  
0   직원의 대면 응대를 활성화해주세요(상담은 사람과 만나서 하고 싶어요, 추천받은 제품...  
22  비대면 응대를 활성화해주세요(낯을 가려요, 집에서 쇼핑하고 싶어요, 혼자 고민하고 ...  
62  직원의 대면 응대를 활성화해주세요(상담은 사람과 만나서 하고 싶어요, 추천받은 제품...

In [96]:
purchaser_genders = dict()
purchaser_genders['여성'] = 0
purchaser_genders['남성'] = 0

for gender in purchasers.iloc[:,2]:
    if gender == '여성':
        purchaser_genders[gender] += 1
    elif gender == '남성':
        purchaser_genders[gender] += 1
    else:
        continue

purchaser_genders

응답을 원하지 않음
응답을 원하지 않음


{'여성': 89, '남성': 31}

In [11]:
ques

['연령대를 선택해주세요. (2020 - "출생년도" + 1의 결과값)',
 '성별을 선택해주세요',
 '기초제품을 구매하거나 선물한 적이 있나요?',
 '평소 화장품에 대한 관심/지식은 어느정도인가요? (복수응답 가능)',
 '기초제품 구매 시, 선택에 어려움을 겪은 적이 있나요? 그 이유를 말씀해주세요. (복수응답가능)',
 '화장품 구매에 있어, 오프라인 매장 방문의 목적이 무엇인가요? (복수응답)',
 '오프라인 매장 방문을 꺼리는 이유는 무엇인가요?',
 '화장품과 관련하여 유튜브로 정보를 탐색하시나요?',
 '화장품과 관련한 어플을 사용하시나요? (화해, 글로우픽 등)',
 '화장품과 관련하여 유튜브 혹은 어플을 사용하는 이유는 무엇인가요?',
 '어떻게 하면 기초 제품 구매가 더 편리할 것 같아요? (복수응답가능)']

In [78]:
# 기초제품 구매에 어려움을 겪는 사람 = 자신의 피부 타입에 맞게 고르지는 못하는 사람들
data['related'] = data[ques[3]].apply(lambda x: 'the related' if re.search('(업계|할 수 있어요)', x) else 'unchoosable')
data.head()

# 기초제품 구매에 어려움을 겪는 사람 = 자신의 피부 타입에 맞게 고르지는 못하는 사람들
df_purchasers['related'] = df_purchasers[ques[3]].apply(lambda x: 'the related' if re.search('(업계|할 수 있어요)', x) else 'unchoosable')
df_purchasers.head()

연령대를 선택해주세요. (2020 - "출생년도" + 1의 결과값) 성별을 선택해주세요 기초제품을 구매하거나 선물한 적이 있나요?  \
0                              20 ~ 24세         여성                       예   
1                              20 ~ 24세         여성                       예   
2                              20 ~ 24세         여성                       예   
3                                19세 이하         여성                       예   
4                              20 ~ 24세         여성                       예   

                평소 화장품에 대한 관심/지식은 어느정도인가요? (복수응답 가능)  \
0  건성, 복합성, 지성이 무엇인지는 알고 있어요.;비타민이 미백에 좋다는 것, 시카와...   
1  비타민이 미백에 좋다는 것, 시카와 티트리 성분이 진정효과가 있다는 것 등을 알고 ...   
2  건성, 복합성, 지성이 무엇인지는 알고 있어요.;비타민이 미백에 좋다는 것, 시카와...   
3                     자신의 피부에 맞게 상품을 비교하고 선택할 수 있어요.   
4  비타민이 미백에 좋다는 것, 시카와 티트리 성분이 진정효과가 있다는 것 등을 알고 ...   

  기초제품 구매 시, 선택에 어려움을 겪은 적이 있나요? 그 이유를 말씀해주세요. (복수응답가능)  \
0  제품이 너무 다양하고 많아서 정보를 알 수 없어서;진짜 좋은 제품인지 알 수 없어서...      
1     제품이 너무 다양하고 많아서 정보를 알 수 없어서;진짜 좋은 제품인지 알 수 없어서      
2     제품이 너무 다양하고 많아서 정보를 알 수 없어서;진짜 좋은 제품인지 알 수 없어서      
3                                 진짜 좋은 제품인지 알 수 없어서      
4  제품이 너무 다양하고 많아서 정보를 알 수 없어서;피부가 민감하지 않아서 제품 다양...      

           화장품 구매에 있어, 오프라인 매장 방문의 목적이 무엇인가요? (복수응답)  \
0  제품을 직접 테스트하고 크기 등을 보려고;사람에게 물어보고 구매하고 싶어서;온라인 ...   
1                             제품을 직접 테스트하고 크기 등을 보려고   
2          제품을 직접 테스트하고 크기 등을 보려고;사람에게 물어보고 구매하고 싶어서   
3                             제품을 직접 테스트하고 크기 등을 보려고   
4         집이랑 가까워서;그냥 집가다가 생각나면 사는 거고 아니면 쿠팡으로 사는거구~   

                          오프라인 매장 방문을 꺼리는 이유는 무엇인가요?  \
0                               오프라인 매장 방문을 꺼리지 않아요!   
1  오프라인 매장 방문을 꺼리지 않아요!;온라인 쇼핑에 익숙해져서;혼자 비교하고 결정하...   
2                       직원들의 접객이 불편해서;혜택이 더 다양하고 많아서   
3                               오프라인 매장 방문을 꺼리지 않아요!   
4                               오프라인 매장 방문을 꺼리지 않아요!   

   화장품과 관련하여 유튜브로 정보를 탐색하시나요?  화장품과 관련한 어플을 사용하시나요? (화해, 글로우픽 등)  \
0                         1.0                                0.0   
1                         0.0                                0.0   
2                         3.0                                1.0   
3                         4.0                                4.0   
4                         1.0                                0.0   

                화장품과 관련하여 유튜브 혹은 어플을 사용하는 이유는 무엇인가요?  \
0         언젠가 구매할지도 모르니까;색조, 메이트업베이스 등에 대한 정보를 얻기 위해   
1                                            사용하지 않음   
2             유행을 알고 싶어서;색조, 메이트업베이스 등에 대한 정보를 얻기 위해   
3  색조, 메이트업베이스 등에 대한 정보를 얻기 위해;기초제품(스킨, 로션, 미스트, ...   
4                             그냥 화장한 얼굴이 예뻐서 대리만족하려구   

              어떻게 하면 기초 제품 구매가 더 편리할 것 같아요? (복수응답가능)      related  
0  직원의 대면 응대를 활성화해주세요(상담은 사람과 만나서 하고 싶어요, 추천받은 제품...  the related  
1  비대면 응대를 활성화해주세요(낯을 가려요, 집에서 쇼핑하고 싶어요, 혼자 고민하고 ...  unchoosable  
2  직원의 대면 응대를 활성화해주세요(상담은 사람과 만나서 하고 싶어요, 추천받은 제품...  unchoosable  
3  특징별로 상품에 대한 요약 정리를 해주세요(보습라인 : 가,나,다 / 여드름 진정 ...  the related  
4  특징별로 상품에 대한 요약 정리를 해주세요(보습라인 : 가,나,다 / 여드름 진정 ...  unchoosable

In [55]:
unchoosable = df_purchasers[df_purchasers['related'] == 'unchoosable']
unchoosable.head()

연령대를 선택해주세요. (2020 - "출생년도" + 1의 결과값) 성별을 선택해주세요 기초제품을 구매하거나 선물한 적이 있나요?  \
1                              20 ~ 24세         여성                       예   
2                              20 ~ 24세         여성                       예   
4                              20 ~ 24세         여성                       예   
5                              20 ~ 24세         남성                       예   
7                              25 ~ 29세         여성                       예   

                평소 화장품에 대한 관심/지식은 어느정도인가요? (복수응답 가능)  \
1  비타민이 미백에 좋다는 것, 시카와 티트리 성분이 진정효과가 있다는 것 등을 알고 ...   
2  건성, 복합성, 지성이 무엇인지는 알고 있어요.;비타민이 미백에 좋다는 것, 시카와...   
4  비타민이 미백에 좋다는 것, 시카와 티트리 성분이 진정효과가 있다는 것 등을 알고 ...   
5                         건성, 복합성, 지성이 무엇인지는 알고 있어요.   
7                         건성, 복합성, 지성이 무엇인지는 알고 있어요.   

  기초제품 구매 시, 선택에 어려움을 겪은 적이 있나요? 그 이유를 말씀해주세요. (복수응답가능)  \
1     제품이 너무 다양하고 많아서 정보를 알 수 없어서;진짜 좋은 제품인지 알 수 없어서      
2     제품이 너무 다양하고 많아서 정보를 알 수 없어서;진짜 좋은 제품인지 알 수 없어서      
4  제품이 너무 다양하고 많아서 정보를 알 수 없어서;피부가 민감하지 않아서 제품 다양...      
5                                 진짜 좋은 제품인지 알 수 없어서      
7  제품이 너무 다양하고 많아서 정보를 알 수 없어서;매번 실패해서;진짜 좋은 제품인지...      

           화장품 구매에 있어, 오프라인 매장 방문의 목적이 무엇인가요? (복수응답)  \
1                             제품을 직접 테스트하고 크기 등을 보려고   
2          제품을 직접 테스트하고 크기 등을 보려고;사람에게 물어보고 구매하고 싶어서   
4         집이랑 가까워서;그냥 집가다가 생각나면 사는 거고 아니면 쿠팡으로 사는거구~   
5          제품을 직접 테스트하고 크기 등을 보려고;사람에게 물어보고 구매하고 싶어서   
7  제품을 직접 테스트하고 크기 등을 보려고;사람에게 물어보고 구매하고 싶어서;집이랑 ...   

                          오프라인 매장 방문을 꺼리는 이유는 무엇인가요?  \
1  오프라인 매장 방문을 꺼리지 않아요!;온라인 쇼핑에 익숙해져서;혼자 비교하고 결정하...   
2                       직원들의 접객이 불편해서;혜택이 더 다양하고 많아서   
4                               오프라인 매장 방문을 꺼리지 않아요!   
5                                     집에서 나가고 싶지 않아서   
7                               오프라인 매장 방문을 꺼리지 않아요!   

   화장품과 관련하여 유튜브로 정보를 탐색하시나요?  화장품과 관련한 어플을 사용하시나요? (화해, 글로우픽 등)  \
1                         0.0                                0.0   
2                         3.0                                1.0   
4                         1.0                                0.0   
5                         1.0                                0.0   
7                         2.0                                0.0   

                화장품과 관련하여 유튜브 혹은 어플을 사용하는 이유는 무엇인가요?  \
1                                            사용하지 않음   
2             유행을 알고 싶어서;색조, 메이트업베이스 등에 대한 정보를 얻기 위해   
4                             그냥 화장한 얼굴이 예뻐서 대리만족하려구   
5             기초제품(스킨, 로션, 미스트, 선케어 등)에 대한 정보를 얻기 위해   
7  색조, 메이트업베이스 등에 대한 정보를 얻기 위해;기초제품(스킨, 로션, 미스트, ...   

              어떻게 하면 기초 제품 구매가 더 편리할 것 같아요? (복수응답가능)      related  
1  비대면 응대를 활성화해주세요(낯을 가려요, 집에서 쇼핑하고 싶어요, 혼자 고민하고 ...  unchoosable  
2  직원의 대면 응대를 활성화해주세요(상담은 사람과 만나서 하고 싶어요, 추천받은 제품...  unchoosable  
4  특징별로 상품에 대한 요약 정리를 해주세요(보습라인 : 가,나,다 / 여드름 진정 ...  unchoosable  
5  특징별로 상품에 대한 요약 정리를 해주세요(보습라인 : 가,나,다 / 여드름 진정 ...  unchoosable  
7  특징별로 상품에 대한 요약 정리를 해주세요(보습라인 : 가,나,다 / 여드름 진정 ...  unchoosable

In [57]:
unchoosable.shape[0]

71

In [75]:
# 스킨 케어 제품을 혼자 결정할 수 없는 사람의 비율
print(f'스킨 케어 제품을 혼자 결정할 수 없는 사람은 전체 응답자 {data.shape[0]}명 중 {unchoosable.shape[0]}명이고, 전체의 약{round(unchoosable.shape[0]*100/data.shape[0], 2)}%이다.')

스킨 케어 제품을 혼자 결정할 수 없는 사람은 전체 응답자 135명 중 71명이고, 전체의 약52.59%이다.


In [61]:
def classify_reason(res, regex, boolean='False'):
    if re.search(regex, res):
        boolean = 'True'
    return boolean

In [63]:
# 혼자 결정할 수 없는 사람들 중 기초제품 구매에 어려움을 겪은 이유가 지식 부족인 비율 알아보기
unchoosable['unchoosable reason'] = unchoosable[ques[4]].apply(lambda x: 'knowledge' if classify_reason(x, '(지식.*(없[어다]|부족)|몰라)') else 'others')
unchoosable.head()

C:\Users\rlath\AppData\Local\Continuum\anaconda3\lib\site-packages\ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  


연령대를 선택해주세요. (2020 - "출생년도" + 1의 결과값) 성별을 선택해주세요 기초제품을 구매하거나 선물한 적이 있나요?  \
1                              20 ~ 24세         여성                       예   
2                              20 ~ 24세         여성                       예   
4                              20 ~ 24세         여성                       예   
5                              20 ~ 24세         남성                       예   
7                              25 ~ 29세         여성                       예   

                평소 화장품에 대한 관심/지식은 어느정도인가요? (복수응답 가능)  \
1  비타민이 미백에 좋다는 것, 시카와 티트리 성분이 진정효과가 있다는 것 등을 알고 ...   
2  건성, 복합성, 지성이 무엇인지는 알고 있어요.;비타민이 미백에 좋다는 것, 시카와...   
4  비타민이 미백에 좋다는 것, 시카와 티트리 성분이 진정효과가 있다는 것 등을 알고 ...   
5                         건성, 복합성, 지성이 무엇인지는 알고 있어요.   
7                         건성, 복합성, 지성이 무엇인지는 알고 있어요.   

  기초제품 구매 시, 선택에 어려움을 겪은 적이 있나요? 그 이유를 말씀해주세요. (복수응답가능)  \
1     제품이 너무 다양하고 많아서 정보를 알 수 없어서;진짜 좋은 제품인지 알 수 없어서      
2     제품이 너무 다양하고 많아서 정보를 알 수 없어서;진짜 좋은 제품인지 알 수 없어서      
4  제품이 너무 다양하고 많아서 정보를 알 수 없어서;피부가 민감하지 않아서 제품 다양...      
5                                 진짜 좋은 제품인지 알 수 없어서      
7  제품이 너무 다양하고 많아서 정보를 알 수 없어서;매번 실패해서;진짜 좋은 제품인지...      

           화장품 구매에 있어, 오프라인 매장 방문의 목적이 무엇인가요? (복수응답)  \
1                             제품을 직접 테스트하고 크기 등을 보려고   
2          제품을 직접 테스트하고 크기 등을 보려고;사람에게 물어보고 구매하고 싶어서   
4         집이랑 가까워서;그냥 집가다가 생각나면 사는 거고 아니면 쿠팡으로 사는거구~   
5          제품을 직접 테스트하고 크기 등을 보려고;사람에게 물어보고 구매하고 싶어서   
7  제품을 직접 테스트하고 크기 등을 보려고;사람에게 물어보고 구매하고 싶어서;집이랑 ...   

                          오프라인 매장 방문을 꺼리는 이유는 무엇인가요?  \
1  오프라인 매장 방문을 꺼리지 않아요!;온라인 쇼핑에 익숙해져서;혼자 비교하고 결정하...   
2                       직원들의 접객이 불편해서;혜택이 더 다양하고 많아서   
4                               오프라인 매장 방문을 꺼리지 않아요!   
5                                     집에서 나가고 싶지 않아서   
7                               오프라인 매장 방문을 꺼리지 않아요!   

   화장품과 관련하여 유튜브로 정보를 탐색하시나요?  화장품과 관련한 어플을 사용하시나요? (화해, 글로우픽 등)  \
1                         0.0                                0.0   
2                         3.0                                1.0   
4                         1.0                                0.0   
5                         1.0                                0.0   
7                         2.0                                0.0   

                화장품과 관련하여 유튜브 혹은 어플을 사용하는 이유는 무엇인가요?  \
1                                            사용하지 않음   
2             유행을 알고 싶어서;색조, 메이트업베이스 등에 대한 정보를 얻기 위해   
4                             그냥 화장한 얼굴이 예뻐서 대리만족하려구   
5             기초제품(스킨, 로션, 미스트, 선케어 등)에 대한 정보를 얻기 위해   
7  색조, 메이트업베이스 등에 대한 정보를 얻기 위해;기초제품(스킨, 로션, 미스트, ...   

              어떻게 하면 기초 제품 구매가 더 편리할 것 같아요? (복수응답가능)      related  \
1  비대면 응대를 활성화해주세요(낯을 가려요, 집에서 쇼핑하고 싶어요, 혼자 고민하고 ...  unchoosable   
2  직원의 대면 응대를 활성화해주세요(상담은 사람과 만나서 하고 싶어요, 추천받은 제품...  unchoosable   
4  특징별로 상품에 대한 요약 정리를 해주세요(보습라인 : 가,나,다 / 여드름 진정 ...  unchoosable   
5  특징별로 상품에 대한 요약 정리를 해주세요(보습라인 : 가,나,다 / 여드름 진정 ...  unchoosable   
7  특징별로 상품에 대한 요약 정리를 해주세요(보습라인 : 가,나,다 / 여드름 진정 ...  unchoosable   

  unchoosable reason  
1          knowledge  
2          knowledge  
4          knowledge  
5          knowledge  
7          knowledge

In [76]:
print(f'{round(unchoosable[unchoosable["unchoosable reason"] == "knowledge"].shape[0]*100/unchoosable.shape[0], 2)}%가 스킨 케어 지식 부족으로 구매에 어려움을 겪은 적이 있다고 말했다.')

100.0%가 스킨 케어 지식 부족으로 구매에 어려움을 겪은 적이 있다고 말했다.
